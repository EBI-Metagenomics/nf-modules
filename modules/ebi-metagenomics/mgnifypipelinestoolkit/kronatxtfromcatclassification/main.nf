process MGNIFYPIPELINESTOOLKIT_KRONATXTFROMCATCLASSIFICATION {
    tag "${meta.id}"
    label 'process_single'

    container 'microbiome-informatics/mgnify-pipelines-toolkit:1.0.3'

    input:
    tuple val(meta), path(cat_output)
    tuple val(meta2), path(taxonomy)

    output:
    tuple val(meta), path("${meta.id}.krona.txt"), emit: krona_txt
    path "versions.yml"                          , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    """
    krona_txt_from_cat_classification \\
        --input ${cat_output} \\
        --output ${meta.id}.krona.txt \\
        --names_dmp ${taxonomy}/names.dmp \\
        --nodes_dmp ${taxonomy}/nodes.dmp

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mgnify-pipelines-toolkit: \$(get_mpt_version)
    END_VERSIONS
    """

    stub:
    """
    touch ${meta.id}.krona.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mgnify-pipelines-toolkit: \$(get_mpt_version)
    END_VERSIONS
    """
}
