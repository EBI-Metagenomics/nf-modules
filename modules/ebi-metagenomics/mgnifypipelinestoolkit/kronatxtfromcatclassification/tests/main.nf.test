nextflow_process {

    name "Test Process MGNIFYPIPELINESTOOLKIT_KRONATXTFROMCATCLASSIFICATION"
    script "../main.nf"
    process "MGNIFYPIPELINESTOOLKIT_KRONATXTFROMCATCLASSIFICATION"

    tag "modules"
    tag "modules_ebimetagenomics"
    tag "mgnifypipelinestoolkit"
    tag "mgnifypipelinestoolkit/kronatxtfromcatclassification"

    test("Run MGNIFYPIPELINESTOOLKIT_KRONATXTFROMCATCLASSIFICATION module - test dataset") {

        when {
            process {
                """
                input[0] = [ [id:'test'], [ file("${moduleDir}/tests/data/cat.tsv", checkIfExists: true) ] ]
                input[1] = [ [id:'test_db'], [ file("${moduleDir}/tests/data/tax", checkIfExists: true) ] ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("Run MGNIFYPIPELINESTOOLKIT_KRONATXTFROMCATCLASSIFICATION module - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [ [id:'test'], [ file("${moduleDir}/tests/data/cat.tsv", checkIfExists: true) ] ]
                input[1] = [ [id:'test_db'], [ file("${moduleDir}/tests/data/tax", checkIfExists: true) ] ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
