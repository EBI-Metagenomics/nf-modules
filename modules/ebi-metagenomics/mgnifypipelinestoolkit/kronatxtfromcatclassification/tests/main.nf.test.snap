{
    "Run MGNIFYPIPELINESTOOLKIT_KRONATXTFROMCATCLASSIFICATION module - test dataset": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.krona.txt:md5,62e1bcaac47b979cafb7364cdc48eafa"
                    ]
                ],
                "1": [
                    "versions.yml:md5,187d47b1098964f9569722bf6572138c"
                ],
                "krona_txt": [
                    [
                        {
                            "id": "test"
                        },
                        "test.krona.txt:md5,62e1bcaac47b979cafb7364cdc48eafa"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,187d47b1098964f9569722bf6572138c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-27T22:24:30.725629"
    },
    "Run MGNIFYPIPELINESTOOLKIT_KRONATXTFROMCATCLASSIFICATION module - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.krona.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,187d47b1098964f9569722bf6572138c"
                ],
                "krona_txt": [
                    [
                        {
                            "id": "test"
                        },
                        "test.krona.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,187d47b1098964f9569722bf6572138c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-27T22:24:33.264533"
    }
}